cocos/ui/UIEditBox/iOS/CCUIMultilineTextField.mm

```diff
@@ -130,6 +127,42 @@ - (void)drawRect:(CGRect)rect
     [super drawRect:rect];
 }
 
+- (void)setInnerPadding:(float)left :(float)top :(float)right :(float)bottom
+{
+    auto glview = cocos2d::Director::getInstance()->getOpenGLView();
+    float factor = glview->getScaleX() / glview->getContentScaleFactor();
+    self.textContainerInset = UIEdgeInsetsMake(top * factor, left * factor, bottom * factor, right * factor);
+    
+    CGRect frame = _placeHolderLabel.frame;
+    frame.origin.x = left * factor;
+    frame.origin.y = top * factor;
+    frame.size.width = self.bounds.size.width - (left * factor) - (right * factor);
+    _placeHolderLabel.frame = frame;
+    
+}
+
+- (void)setTextAlignment:(cocos2d::TextHAlignment)hAlign :(cocos2d::TextVAlignment)vAlign
+{
+    switch (hAlign) {
+        case cocos2d::TextHAlignment::LEFT :
+            [self setTextAlignment:NSTextAlignmentLeft];
+            [_placeHolderLabel setTextAlignment:NSTextAlignmentLeft];
+            break;
+        case cocos2d::TextHAlignment::CENTER :
+            [self setTextAlignment:NSTextAlignmentCenter];
+            [_placeHolderLabel setTextAlignment:NSTextAlignmentCenter];
+            break;
+        case cocos2d::TextHAlignment::RIGHT :
+            [self setTextAlignment:NSTextAlignmentRight];
+            [_placeHolderLabel setTextAlignment:NSTextAlignmentRight];
+            break;
+        default:
+            break;
+    }
+    
+    _verticalAlign = vAlign;
```

Comment:
@anjandaffo Shouldn't you refactor the code to update the vertical alignment by moving it outside `observeValueForKeyPath:...` into a separate method, then call it from here to make sure that it gets updated?

An easy fix would be to move the code from `observeValueForKeyPath:` into this method, then within `observeValueForKeyPath:...` call something like:

```objc
+-(void)observeValueForKeyPath:(NSString *)keyPath   ofObject:(id)object   change:(NSDictionary *)change   context:(void *)context {
    // update vertical insets
    [self setTextAlignment:_horizontalAlignment :_verticalAlignment];
}
```

A much better way is to have a separate method, call it `updateVerticalAlignment`, and it gets called from here and `observeValueForKeyPath:...`